static/app/views/nav/secondary/sections/issues/issueViews/useStarredIssueViews.tsx

```diff
@@ -0,0 +1,58 @@
+import {useCallback} from 'react';
+
+import {getApiQueryData, setApiQueryData, useQueryClient} from 'sentry/utils/queryClient';
+import useOrganization from 'sentry/utils/useOrganization';
+import {makeFetchStarredGroupSearchViewsKey} from 'sentry/views/issueList/queries/useFetchStarredGroupSearchViews';
+import type {StarredGroupSearchView} from 'sentry/views/issueList/types';
+import type {NavIssueView} from 'sentry/views/nav/secondary/sections/issues/issueViews/issueViewNavItems';
+
+export function useStarredIssueViews() {
+  const organization = useOrganization();
+  const queryClient = useQueryClient();
+
+  const groupSearchViews = getApiQueryData<StarredGroupSearchView[]>(
```

Comment:
This won't cause a rerender when the data changes, so it's better to use the hook. You can pass a prop so that it only updates when the data changes if you want:

```tsx
const {data: groupSearchViews} = useApiQuery(key, {notifyOnChangeProps: ['data']})
```